### ISSUE TITLE
Feature request: Discover subnet nodes

### ISSUE THREAD

**Issue by mikkelam** [tailscale/tailscale#1482]
**Is your feature request related to a problem? Please describe.**
I use tailscale's subnet routing extensively and love it, but i find it quite painful having to discover subnet nodes, combining that with the fact that u cant use magicDNS to access them. 

**Describe the solution you'd like**
I think it would be cool if tailscale could show these next to all other nodes in the admin panel and perhaps the GUI.

**Describe alternatives you've considered**

i've been using port scanning software like nmap

**Comment by rosszurowski** [comment]
@mikkelam can you explain what you mean by "discover subnet nodes"? Is this about discovering which devices in your network already expose subnet routes over Tailscale?

If so, you should see a "subnet" badge on the machines page which can help you locate them. 

<img width="551" alt="image" src="https://user-images.githubusercontent.com/303731/110965322-1175ed80-8322-11eb-9026-4c3aa69d5f1c.png">

We also have a (currently undocumented, needing more UI) search filter feature that can help find all subnet nodes in your network. If you [type `has:subnets` into the machine page search box](https://login.tailscale.com/admin/machines?q=has%3Asubnets), you'll see all devices exposing subnet routes.

**Comment by mikkelam** [comment]
Thank you for your reply @rosszurowski. Sorry about any confusion. I mean that I would like to have some way of viewing  the IP of the nodes, perhaps along with their hostnames, on the actual subnet. So instead of saying that it has subnets, it should tell me what are the nodes on that subnet.


 Does that make sense?

**Comment by rosszurowski** [comment]
Ah, makes sense. I can see how that'd be useful! I'm not sure if that's possible — my understanding is subnet relays just blindly forward packets within the ranges they're told too, but maybe @bradfitz has thoughts here.

**Comment by DentonGentry** [comment]
We might be able to do things using mDNS/DNS-SD/SLP to discover nodes on local subnets and some details of services they offer. That would be a relatively large effort.

**Comment by DentonGentry** [comment]
Copying from @daderson on https://github.com/tailscale/tailscale/issues/1252

----

WebRTC exchanges candidate endpoints similarly to magicsock, for NAT traversal. However, it's implemented a very cool idea to preserve peer privacy for private LAN endpoints: https://tools.ietf.org/html/draft-ietf-rtcweb-mdns-ice-candidates-04

TL;DR: instead of peers telling each other "I might be reachable on 192.168.0.15:1234" (which leaks private LAN information), nodes register a UUID hostname in their local mdns resolver (e.g. `<uuid4>.local -> 192.168.0.15:1234`), then give their peer the UUID. The peer can then query their local mdns for that UUID, and if they're on the same LAN as their peer, they'll learn the ip:port that they can use.

This is pretty great because it hides information from peers that are not in a position to make use of that information, while still offering a way to upgrade to a direct LAN connection.

We should do that as well. There may be complications for things like communication between a Docker container and its host (which are LAN IPs, but mdns seems less likely to work on that link), and this functionality depends on nodes being able to register new names in their local mdns resolver, which may not be true on all OSes.

**Comment by bradfitz** [comment]
Note that iOS has restrictions around local LAN use. I'm not sure whether our app's existing "entitlements" cover this.

* https://thomask.sdf.org/blog/2021/08/24/apples-bizarre-crackdown-on-multicast.html
* https://developer.apple.com/forums/thread/663858

/cc @crawshaw @nickoneill

**Comment by DentonGentry** [comment]
Substantially similar to https://github.com/tailscale/tailscale/issues/1013

**Comment by DentonGentry** [comment]
Substantially similar to https://github.com/tailscale/tailscale/issues/814

**Comment by ganduulgag** [comment]
is this feature going to be implemented near in the future?
